### ISSUE TITLE
mysql errors are missing properties

### ISSUE THREAD

**Issue by M-TGH** [flowtype/flow-typed#2039]
Currently the errors within the mysql package are merely typed as the default Error (it does extend these so it's not inherently wrong); [here](https://github.com/flowtype/flow-typed/blob/18b77b0f2e00a6ef8990345a1f181e490d856cdc/definitions/npm/mysql_v2.x.x/flow_v0.28.x-/mysql_v2.x.x.js#L133).

However the mysql package adds a [couple of properties](https://github.com/mysqljs/mysql#error-handling) for debugging/logging purposes. As of [flow@v0.68.0](https://github.com/facebook/flow/releases/tag/v0.68.0) this probably became a bit bigger of a problem as they no longer allow unknown property checks.

Am willing to open a PR however I'm not completely certain how to go about adding this, was thinking of something along the lines of adding a custom error type:
```
declare type MySQLError = Error & {
    code: string,
    fatal: boolean,
    sql?: string,
    sqlMessage?: string,
}
```
Or since it's probably a class that extends it:
```
declare class MySQLError extends Error{
    code: string,
    fatal: boolean,
    sql?: string,
    sqlMessage?: string,
}
```

**Comment by GAntoine** [comment]
Extending the class is most likely the way to go.

**Comment by AndrewSouthpaw** [comment]
Yeah, I think the `Foo & Bar` approach would work better if `Error` were a type, not a class itself.
